yarn-project/aztec-nr/aztec/src/oracle/get_block_data.nr

```diff
@@ -0,0 +1,36 @@
+use dep::std::merkle::compute_merkle_root;
+use crate::constants_gen::{
+    HISTORIC_BLOCK_DATA_LENGTH,
+    HISTORIC_BLOCKS_TREE_HEIGHT,
+};
+use crate::abi::HistoricBlockData;
+use crate::oracle::get_membership_witness::{
+    get_membership_witness,
+    MembershipWitness,
+};
+
+#[oracle(getBlockData)]
+fn get_block_data_oracle(_block_number: Field) -> [Field; HISTORIC_BLOCK_DATA_LENGTH] {}
+
+unconstrained pub fn get_block_data_internal(block_number: Field) -> HistoricBlockData {
+    let block_data = get_block_data_oracle(block_number);
+    HistoricBlockData::deserialize(block_data)
+}
+
+pub fn get_block_data(block_number: Field, current_blocks_tree_root: Field) -> HistoricBlockData {
```

Comment:
Think it would easier for devs to use this correctly if they don't have to pass in the "magic" but specific value for `current_blocks_tree_root`. Seems better to forward the context and then get it from there instead to cause less specific user import needed.

Reply:
Good point. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/3255/commits/acc1cb1730b8421c3f3f7c417d9b20727102d265